### ISSUE TITLE
CoffeeScript 2 support

### ISSUE THREAD

**Issue by laurentpayot** [cypress-io/cypress#663]
- Operating System: Ubuntu 17.04
- Cypress Version: 0.20.1
- Browser Version: Chrome 62, Chromium 60, Electron 53

### Is this a Feature or Bug?
Feature

### Current behavior:
Cypress does not support [CoffeeScript 2](http://coffeescript.org/announcing-coffeescript-2/) syntax (especially the `await` keyword), that was released a few days ago.

To fix this in Cypress server I replaced CS1 with CS2 (see below how to reproduce).
Then everything was working fine with my CS2 tests, but only using the Cypress app (via `cypress open`). Via `cypress run` I got weird errors with electron, chrome and chromium browsers (see stack trace below ).

### Desired behavior:
CoffeeScript 2 tests working "out of the box" after a fresh Cypress install, with `cypress open` as well as `cypress run`.

### How to reproduce:
In Cypress server I replaced CS1 with CS2 like so:
```shell
cd node_modules/cypress/dist/Cypress/resources/app/packages/server/
npm i coffeescript@next
cd node_modules
mv coffee-script/ coffee-script_V1
ln -s coffeescript/ coffee-script
```

### Additional Info (images, stack traces, etc)
Using `cypress run` with CoffeeScript 2, I get the following error:
```shell
npx cypress run --browser electron --config baseUrl=http://localhost:5000

Started video recording: /home/laurent/myproject/cypress/videos/k38ej.mp4

  (Tests Starting)


  1) An uncaught error was detected outside of a test

  0 passing (236ms)
  1 failing

  1)  An uncaught error was detected outside of a test:
     Uncaught SyntaxError: missing ) after argument list

Cypress could not associate this error to any specific test.

We dynamically generated a new test to display this failure.
```

**Comment by brian-mann** [comment]
Can you provide an example of your test code?

**Comment by laurentpayot** [comment]
Sure.
```coffeescript
describe("dummy tests:", ->

  context("foo:", ->

    it.only "should return 4", ->
      # result = 2 + 2 # no error
      result = await Promise.resolve(2 + 2) # generates an error with `cypress run`
      expect      result
      .to.equal    4

  )
)
```

**Comment by brian-mann** [comment]
Just curious - are you writing unit tests in Cypress? Why would you use the `await` syntax with Cypress tests?

**Comment by laurentpayot** [comment]
Yes I write "not so unit" tests in Cypress. I use Firebase and I need to test some promise-returning CRUD functions I wrote that are using Firebase calls. That way I can test database rules located on the Firebase cloud side.
All these CRUD tests do not need UI so I was simply using mocha, but I moved them inside Cypress because:
1. it is nice to have _all_ the tests in a single reporter
2. mocha CLI test environment was node, not the browser as with Cypress (am I right?) 

Also to use Firebase node admin functionalities (test users creation on the fly etc.) I had to set an express server up that I'm calling inside Cypress via POST fetch requests . Let me know if you see a better alternative…

**Comment by brian-mann** [comment]
Transpiling to CS2 will soon be supported by plugins / extensions API

**Comment by laurentpayot** [comment]
#684 is really epic. Cypress is getting better and better :+1:

**Comment by brian-mann** [comment]
You can now do this yourself by modifying the `browserify` options and add your own coffeescript 2 support.

Fixed by https://github.com/cypress-io/cypress/pull/888.

**Comment by brian-mann** [comment]
Fixed in [`1.1.0`](https://docs.cypress.io/guides/references/changelog.html#1-1-0).
